src/jestUtils.ts

```diff
@@ -0,0 +1,270 @@
+import { fireEvent } from '@testing-library/react-native';
+
+export const fireGestureHandlerEvent = (
+  component: any,
+  name: string,
+  config: any
+) => {
+  fireEvent(component, name, config);
+};
+
+export interface TapConfig {
+  x?: number;
+  y?: number;
+  absoluteX?: number;
+  absoluteY?: number;
+}
+
+export interface PanConfig extends TapConfig {
+  translationX?: number;
+  translationY?: number;
+  velocityX?: number;
+  velocityY?: number;
+}
+
+export interface LongPressConfig extends TapConfig {
+  duration?: number;
+}
+
+export interface RotationConfig {
+  rotation?: number;
+  velocity?: number;
+  anchorX?: number;
+  anchorY?: number;
+}
+
+export interface PinchConfig {
+  scale?: number;
+  velocity?: number;
+  focalX?: number;
+  focalY?: number;
+}
+
+const sendBeginEvent = (component: any, eventData?: Record<string, any>) => {
+  // simulate undetermined -> begin state change
+  fireEvent(component, 'gestureHandlerStateChange', {
+    nativeEvent: {
+      oldState: 0,
+      state: 2,
+      ...eventData,
+    },
+  });
+};
+
+const sendProgressEvent = (component: any, eventData?: Record<string, any>) => {
+  // simulate begin -> active state change
+  fireEvent(component, 'gestureHandlerStateChange', {
+    nativeEvent: {
+      oldState: 2,
+      state: 4,
+      ...eventData,
+    },
+  });
+};
```

Comment:
This method currently sends gesture updates in `gestureHandlerStateChange` events instead of `gestureHandlerEvent`. It should be split in two: one to simulate `BEGAN` -> `ACTIVE` state change, the other to simulate sending updates in the `ACTIVE` state.
```suggestion
const sendActivateEvent = (component: any, eventData?: Record<string, any>) => {
  // simulate begin -> active state change
  fireEvent(component, 'gestureHandlerStateChange', {
    nativeEvent: {
      oldState: 2,
      state: 4,
      ...eventData,
    },
  });
};

const sendProgressEvent = (component: any, eventData?: Record<string, any>) => {
  // simulate gesture update
  fireEvent(component, 'gestureHandlerEvent', {
    nativeEvent: {
      state: 4,
      ...eventData,
    },
  });
};
```